### DISCUSSION TITLE
Problem when using motion filament runout sensor

### DISCUSSION THREAD

**Discussion by Horgul123** [Guilouz/Creality-Helper-Script-Wiki#495] [Issues]
The M600 support moves the filament sensor config into the M600 script.
When using a motion runout sensor (BIGTREETECH Smart Filament Sensor V2.0), which has a different config in the printer.cfg, installing the M600 support results in 2 runout sensor configs which prevents the printer to start (duplicate use of pin PC15).
When removing the config in the M600 scripts the repro for the helper script becomes dirty.

Is it really nessecary to move the filament sensor config away from the printer.cfg?

Config for the filament sensor:
`[filament_motion_sensor filament_sensor]
switch_pin: PC15
detection_length: 4.5
extruder: extruder
pause_on_runout: true
runout_gcode:
  M117 Filament encoder runout
  {% if printer.extruder.can_extrude|lower == 'true' %}
   G91
   G0 E30 F600
   G90
 {% endif %}
insert_gcode:
 M117 Filament encoder inserted
`

**Comment by Guilouz** [comment]
The script disable [filament_switch_sensor filament_sensor] config in printer.cfg when you install M600 support.

**Reply by Horgul123** [reply]
And adds a "wrong" filament config in the M600 script.

Why? there is a working filament sensor in the printer.cfg. 

And: How can I clean the "dirty repo"?

**Reply by Guilouz** [reply]
It’s not wrong and it have been fixed to call M600 when filament sensor is triggered.

Wich dirty repo?

**Reply by Horgul123** [reply]
Maybe you should add a check if another filament senor is defined. At the moment installing the M600 script leads to an error with another sensor and it takes a little knowledge to identfiy and fix the cause. And if you alter the M600 scripts you get this error. (deleted the M600 meanwhile, at the beginning it told be, that the M600 script was changed.

![2024-03-28_14-21-38](https://github.com/Guilouz/Creality-K1-Series/assets/45976162/ec84af58-158f-45eb-9ae9-6698cec378fa)

**Reply by Guilouz** [reply]
On k1 max there is 2 filament sensor config and the second one not integere with the one in M600 config.

If you have modified files in helper script it’s normal you have a dirty repo.

**Reply by Horgul123** [reply]
The smart filament senor ist connected to PC15 instead of the original one. so as shown above in the config you need to address PC15. And in the M600 script there is the original sensor with PC15 configured. This leads to the problem.

Is there a chance to "clean" the repo (without M600 support, I removed this via your script)

**Reply by Guilouz** [reply]
Script is done for stock machine not modified one.
Sensor PIN is PC15 so it can’t be changed.

just use repair fonction on web interface.

**Reply by Horgul123** [reply]
I understand, that you can't change it. But maybe you could add a check or a warning. 

Repair worked. Thanks.

**Comment by ViniSug** [comment]
@Horgul123 
To resolve the error message of duplicated use os pins, just add PC15 on [duplicate_pin_override] section

**Reply by Horgul123** [reply]
thanks for the hint. will check this option

**Comment by ViniSug** [comment]
@Horgul123 
I bought the same BTT SFS 2.0.
I know that K1 has 2 pins to connect the runout sensor: PA15 (on the motherboard) and PA10 (on the tool head)
Can you share how you connected the sensors? Did you connect both or just one?

**Reply by Horgul123** [reply]
I'm just using the motion sensor. Works pretty well :)

Just replaced the existing sensor

**Reply by FaintandFuzzies** [reply]
@Horgul123 What does your Klipper code look like?  I keep getting  an error:  The value 'filament_sensor' is not valid for SENSOR. My code looks like this:

[filament_motion_sensor motion_sensor]
switch_pin: PC15
detection_length: 7
extruder: extruder
pause_on_runout: false
runout_gcode:
  PAUSE # [pause/resume] is required for printer.cfg
  M117  Filament Encoder Runout
insert_gcode:
  M117  Filament Encoder Inserted

[filament_switch_sensor filament_sensor_2]
pause_on_runout: false
switch_pin: !nozzle_mcu:PA10

I chose to connect the SFS V2 motion sensor to the main board instead of the switch sensor since you can only choose 1 on the K1 (non-Max)

**Reply by Horgul123** [reply]
I removed the sensor in the meantime. Too many false errors

**Comment by CyberWop** [comment]
Hello, has anyone successfully managed to use the SFS2.0 with a K1MAX ? I have replaced just the original sensor (green to D, red and black, blue not used) not connected anything to motherboard or tool head. But I can't get it to work, I've tried every possible combination in printer.cfg: either I get an error message, or detects break when not supposed to. I've read of people successfully doing it on K1, but their config does not work on my K1max

**Reply by tpruvot** [reply]
Look like you need to use the blue wire... the SFSv2 has 2 switches (for 2 inputs), one for the runout, and one which make a tick each 3mm. Both inputs are pulled up on the BTT manta (gnd on signal so, but unsure if its the presence or the missing filament which send the signal)... 
![image](https://github.com/user-attachments/assets/55cbd0d8-b447-4384-8c49-38ae96c10ac8)

So ignore the green wire first, and when its ok you can try to find a free input pin to configure it... But you need the 3 other wires (5v gnd and switch ones)

**Reply by tpruvot** [reply]
so i finally opened my nozzle cover to double check the availables pins... there is indeed PA10 but also PA8 & PA9 made for a bltouch.. both on rear pcb but not used on the K1Max, the Lidar has its own board/wires & connector
![Nozzle_Board_B](https://github.com/user-attachments/assets/433db549-0ed5-4f46-8f0b-917001411b23)

but all these tiny connectors are not usual, need the right ones or to solder directly on the "backside" which is not accessible due to the double PCB system

**Reply by tpruvot** [reply]
```
#[filament_switch_sensor nozzle_filament_sensor]
#pause_on_runout: true
#switch_pin: !nozzle_mcu:PA10

[filament_switch_sensor SFSv2_runout]
pause_on_runout: false
switch_pin: ^nozzle_mcu:PA8
runout_gcode:
  M117 SFSv2 runout
  PAUSE
insert_gcode:
  M117 SFSv2 inserted

[filament_motion_sensor SFSv2_encoder]
switch_pin: ^nozzle_mcu:PA9
detection_length: 2.88
extruder: extruder
pause_on_runout: false
runout_gcode:
  M117 SFSv2 encoder runout
  PAUSE
insert_gcode:
  M117 SFSv2 encoder resume
```

to try.... if you find the 5 pins connector
